src/TomochainNameService.js

```diff
@@ -0,0 +1,31 @@
+"use strict"
+import {
+  registryInit,
+  getResolver,
+} from "./helper/tns/registryService";
+import {
+  resolverInit,
+  getAddress
+} from "./helper/tns/resolverService";
+class TomochainNameService {
+  // Provider URL
+  constructor(restURL, networkId) {
+    this.restURL = restURL;
+    this.networkId = networkId;
+  }
+
+  async getResolver(name) {
+    // TODO using web3 to fetch data
+    registryInit(this.restURL, this.networkId);
+    const resolverAddr = await getResolver(name);
+    return resolverAddr;
+  }
+
+  async getAddress(name, resolverAddr) {
```

Comment:
This function can use more easily just fill in `name`, we can get the `resolver` from `getResolver` function and response the `address`